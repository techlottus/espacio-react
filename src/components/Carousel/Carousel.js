import React, { useEffect } from 'react'



export const Carousel = React.memo(({data, onSlide})  => {
    const carouselRef = React.createRef();

    useEffect(() => {
        carouselRef.current.data = {
            slides: data.slides || [],
            width: '100%',
            height: window.innerWidth < 991 ? '180px':'338px',
            size: data.size
        }
    },[data]) // eslint-disable-line react-hooks/exhaustive-deps

    useEffect(() => {
        carouselRef.current.addEventListener('onSlide', onSlide);
    },[]) // eslint-disable-line react-hooks/exhaustive-deps

    return(
        <>
        <lottus-carousel ref={carouselRef}></lottus-carousel>
        </>
    )
})