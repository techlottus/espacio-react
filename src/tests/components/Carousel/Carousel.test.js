import React from "react";
import { mount } from "enzyme";
import { Carousel } from "../../../components/Carousel/Carousel";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-dom/test-utils";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let initState;
let data = {
  slides: null,
  width: null,
  height: null,
  size: null,
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Carousel data={data} />
    </Provider>
  );
};

describe("Test <Carousel />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should put width to 180px", () => {
    data = {
      ...data,
      height: '180px'
    }
    window.innerWidth = 990;

    const wrapper = handleMount(initState);

    wrapper.update();

    window.dispatchEvent(new Event("resize"));

    expect(wrapper.props().children.props.data.height).toBe("180px");
  });
});
