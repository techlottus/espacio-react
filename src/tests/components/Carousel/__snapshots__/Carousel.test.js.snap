// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Carousel /> should init component 1`] = `<lottus-carousel />`;
